tests/melee_dodge_hit_test.cpp

```diff
@@ -55,8 +55,20 @@ static float dodge_with_effect( Creature &critter, const std::string &effect_nam
 static float dodge_wearing_item( avatar &dummy, item &clothing )
 {
     // Get nekkid and wear just this one item
-    std::list<item> temp;
-    while( dummy.takeoff( dummy.i_at( -2 ), &temp ) ) {}
+
+    while( true ) {
+        item &it = dummy.i_at( -2 );
+
+        //If the item is a null pointer then we don't need to take it off.
+        if( it.is_null() ) {
+            break;
+        }
+
+        if( !dummy.takeoff( item_location( *dummy.as_character(), &it ) ) ) {
+            break;
+        }
+    }
```

Comment:
```suggestion
    std::list<item> temp;
    while( true ) {
        item &it = dummy.i_at( -2 );

        //If the item is a null item then we don't need to take it off.
        if( it.is_null() ) {
            break;
        }

        if( !dummy.takeoff( item_location( *dummy.as_character(), &it ), &temp ) ) {
            break;
        }
    }
```
If `temp` is not provided, `takeoff` ask the player to handle the item, which calls the input code which leads to the error in the test.

Reply:
Oh wow, that makes sense now why it was named `temp`. Do you know why that item list is named `res` inside `player::takeoff()`?

Reply:
I think it's short for `result`? It returns the items instead of adding them to the player's inventory.